src/main/java/com/ixortalk/organization/api/rest/OrganizationRestResource.java

```diff
@@ -48,45 +48,42 @@
 
     @Override
     @PreAuthorize("permitAll()")
-    @PostFilter("hasRole('ROLE_ADMIN') or hasRole(filterObject.role)")
+    @PostFilter("hasRole('ROLE_ADMIN') or @securityService.isAdminOfOrganization(filterObject)")
     Collection<Organization> findAll();
 
     @Override
     @PreAuthorize("permitAll()")
-    @PostFilter("hasRole('ROLE_ADMIN') or hasRole(filterObject.role)")
+    @PostFilter("hasRole('ROLE_ADMIN') or @securityService.isAdminOfOrganization(filterObject)")
     Iterable<Organization> findAll(Sort sort);
 
     @Override
     @PreAuthorize("permitAll()")
-    @Query("select o from Organization o where ?#{ hasRole('ROLE_ADMIN') } = true or o.role in ?#{ @securityService.authorities }")
+    @Query("from Organization o where ?#{ hasRole('ROLE_ADMIN') } = true or exists " +
+            "(from Organization o2 join o2.users u where (u.login = ?#{ @userEmailProvider.currentUsersEmail.orElse(null) } and u.isAdmin = true and o2.id = o.id)) ")
     Page<Organization> findAll(Pageable pageable);
 
     @Override
     @PreAuthorize("permitAll()")
-    @PostAuthorize("hasRole('ROLE_ADMIN') or (returnObject.present && hasRole(returnObject.get().role))")
+    @PostAuthorize("hasRole('ROLE_ADMIN') or (returnObject.present && @securityService.isAdminOfOrganization(returnObject))")
```

Comment:
Why do we have the present check here too? For returning `403` instead of `404`?

Reply:
Exactly, if the is present check is omitted, the whenOrganizationDoesNotExist of OrganizationRestController_UsersUsed_IntegrationAndRestDocTest would brake

Reply:
Ok 👍